packages/vow/src/watch-utils.js

```diff
@@ -0,0 +1,106 @@
+// @ts-check
+
+import { M } from '@endo/patterns';
+
+const VowShape = M.tagged(
+  'Vow',
+  harden({
+    vowV0: M.remotable('VowV0'),
+  }),
+);
+
+/**
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {import('./watch.js').Watch} watch
+ * @param {() => import('./types.js').VowKit<any>} makeVowKit
+ */
+export const prepareWatchUtils = (zone, watch, makeVowKit) => {
+  const detached = zone.detached();
+  const makeWatchUtilsKit = zone.exoClassKit(
+    'WatchUtils',
+    {
+      utils: M.interface('Utils', {
+        all: M.call(M.arrayOf(VowShape)).returns(M.any()),
+      }),
+      helpers: M.interface('Helpers', {
+        check: M.call(M.any(), M.any()).returns(),
+      }),
+      watcher: M.interface('Watcher', {
+        onFulfilled: M.call(M.any()).rest(M.any()).returns(M.any()),
+        onRejected: M.call(M.any()).rest(M.any()).returns(M.any()),
+      }),
+    },
+    () => {
+      return {
+        id: 0n,
+        /** @type {MapStore<bigint, number>} */
+        count: detached.mapStore('count'),
+
+        /** @type {MapStore<bigint, number>} */
+        length: detached.mapStore('length'),
+
+        /** @type {MapStore<bigint, any[]>} */
+        results: detached.mapStore('results'),
+
+        /** @type {MapStore<bigint, import('./types.js').VowKit>} */
+        kits: detached.mapStore('kits'),
+      };
```

Comment:
I'd suggest grouping these stores together... I think the overhead is less that way:
```suggestion
    () => {
      /**
        * @typedef {object} VowState
        * @property {number} remaining
        * @property {MapStore<number, any>} resultMap
        * @property {import('./types.js').VowKit['resolver']} resolver
        */
        /** @type {MapStore<bigint, VowState>} */
        const idToVowState = detached.mapStore('idToVowState');
        return {
          nextId: 0n,
          idToVowState,
        };
      }  
```